components/admin_console/filter/filter.tsx

```diff
@@ -0,0 +1,247 @@
+        };
+
+        this.filterRef = React.createRef();
+        this.buttonRef = React.createRef();
+    }
+
+    componentDidMount = () => {
+        document.addEventListener('mousedown', this.handleClickOutside);
+    }
+
+    componentWillUnmount = () => {
+        document.removeEventListener('mousedown', this.handleClickOutside);
+    }
+
+    handleClickOutside = (event: MouseEvent) => {
+        if (this.filterRef?.current?.contains(event.target as Node)) {
+            return;
+        }
+        this.hidePopover();
+    }
+
+    hidePopover = () => {
+        this.setState({show: false});
+        this.buttonRef?.current?.blur();
+    }
+
+    togglePopover = () => {
+        if (this.state.show) {
+            this.hidePopover();
+            return;
+        }
+
+        this.setState({show: true});
+    }
+
+    updateValues = async (values: FilterValues, optionKey: string) => {
+        const options = {
+            ...this.state.options,
+            [optionKey]: {
+                ...this.state.options[optionKey],
+                values: {
+                    ...values
+                },
+            },
+        };
+        this.setState({options, optionsModified: true});
+    }
+
+    onFilter = () => {
+        this.props.onFilter(this.state.options);
+        this.setState({optionsModified: false, show: false, filterCount: this.calculateFilterCount()});
+    }
+
+    calculateFilterCount = () => {
+        const options = this.state.options;
+        let filterCount = 0;
+        this.props.keys.forEach((key) => {
+            const {values, keys} = options[key];
+            keys.forEach((filterKey: string) => {
+                if (values[filterKey].value instanceof Array) {
+                    filterCount += (values[filterKey].value as string[]).length;
+                } else if (values[filterKey].value) {
+                    filterCount += 1;
+                }
+            });
+        });
+        return filterCount;
+    }
+
+    resetFilters = () => {
+        this.setState({options: {...this.props.options}}, this.onFilter);
+    }
+
+    renderFilterOptions = () => {
+        const {keys, options} = this.state;
+        return keys.map((key: string) => {
+            const filter = options[key];
+            const FilterListComponent = filter.type || FilterList;
+
+            return (
+                <FilterListComponent
+                    option={filter}
+                    optionKey={key}
+                    updateValues={this.updateValues}
+                    key={key}
+                />
+            );
+        });
+    }
+
+    render() {
+        const filters = this.renderFilterOptions();
+        const {filterCount} = this.state;
+
+        return (
+            <div
+                className='Filter'
+                ref={this.filterRef}
+            >
+                <button
+                    className={this.state.show ? 'Filter_button Filter__active' : 'Filter_button'}
```

Comment:
3/5 prefer the use of `classnames`

Reply:
Didnt know about this, is it preferable to use classnames even when not dealing with conditional statements?

Reply:
It's not a big deal but its good for consistency so then its easier to introduce 👍